protocol/x/clob/memclob/memclob.go

```diff
@@ -676,53 +675,41 @@ func (m *MemClobPriceTimePriority) PlacePerpetualLiquidation(
 		return 0, 0, nil, err
 	}
 
-	// Skip checking if the account needs to be deleveraged if the liquidation order was partially
-	// or fully-filled.
-	if liquidationOrderStatus.OrderOptimisticallyFilledQuantums > 0 {
-		return liquidationOrderStatus.OrderOptimisticallyFilledQuantums,
-			liquidationOrderStatus.OrderStatus,
-			offchainUpdates,
-			err
-	}
-
-	canPerformDeleveraging, deleverageErr := m.clobKeeper.CanDeleverageSubaccount(ctx, liquidationOrder.GetSubaccountId())
-	if deleverageErr != nil {
-		return 0, 0, offchainUpdates, deleverageErr
-	}
-
-	// Early return to skip deleveraging if the subaccount can't be deleveraged.
-	if !canPerformDeleveraging {
-		return liquidationOrderStatus.OrderOptimisticallyFilledQuantums,
-			liquidationOrderStatus.OrderStatus,
-			offchainUpdates,
-			err
-	}
-
-	// Deleverage the full liquidation order size from the subaccount's position size.
-	deltaQuantums := liquidationOrder.GetBaseQuantums().ToBigInt()
-	if !liquidationOrder.IsBuy() {
-		deltaQuantums = deltaQuantums.Neg(deltaQuantums)
-	}
+	return liquidationOrderStatus.OrderOptimisticallyFilledQuantums,
+		liquidationOrderStatus.OrderStatus,
+		offchainUpdates,
+		err
+}
 
-	fills, _ := m.clobKeeper.OffsetSubaccountPerpetualPosition(
+// DeleverageSubaccount will deleverage a subaccount by finding perpetual positions that can be used to offset
+// the offendign subaccount. All position will be closed at the bankruptcy price of the subaccount that is being
+// deleveraged.
+func (m *MemClobPriceTimePriority) DeleverageSubaccount(
+	ctx sdk.Context,
+	subaccountId satypes.SubaccountId,
+	perpetualId uint32,
+	deltaQuantums *big.Int,
+) (
+	quantumsDeleveraged *big.Int,
+	err error,
```

Comment:
I think it's impossible for this function to return something that's not nil here?

Reply:
yup - added err here for consistency with the liquidation methods.

I am planning to do some cleanup and potentially change some of the panics in this code path to error, so added err here in this PR